### PR TITLE
[MONIT-28397] Discrepancy in buffer file data and proxy backlog charts

### PR DESCRIPTION
**Pull request by laullon**

### CODE REVIEW

File: proxy/src/main/java/com/wavefront/agent/queueing/QueueController.java [hunk 1]
```diff
@@ -51,10 +47,10 @@
   @SuppressWarnings("UnstableApiUsage")
   protected final RateLimiter reportRateLimiter = RateLimiter.create(0.1);
 
-  private AtomicLong currentWeight = null;
-  private final AtomicInteger queueSize = new AtomicInteger();
+  private long currentWeight;
+  private int queueSize;
```
Comment (jbau) [inline proxy/src/main/java/com/wavefront/agent/queueing/QueueController.java:51]: do we think it's safe to remove the atomic?

**Review by jbau — COMMENTED** [review]
would like to briefly discuss why it's safe to remove atomic, otherwise i think it looks ok.

File: proxy/src/main/java/com/wavefront/agent/queueing/QueueController.java (hunk 1 above)
Comment (laullon) [reply proxy/src/main/java/com/wavefront/agent/queueing/QueueController.java:51]: Yes, we create just one instance of this class for each "port", they are private members, update only on thread.
I don't see the need for Atomic operations.

Comment (laullon) [reply proxy/src/main/java/com/wavefront/agent/queueing/QueueController.java:51]: but can be put Atomic back if you think is best

Comment (jbau) [reply proxy/src/main/java/com/wavefront/agent/queueing/QueueController.java:51]: The metric reporting runs in a different thread, actually.  Although that thread does is read the int.  So if only one thread is writing we are still ok.  I'm fine keeping it int.

**Review by jbau — APPROVED** [review]

**Pull request merged**